gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -421,14 +421,16 @@ public EntityFetcherResponse getTimeAggregatedMetrics(
               QueryAndGatewayDtoConverter.convertQueryValueToGatewayValue(
                   row.getColumn(idColumns.size()));
           if (value.getValueType() == ValueType.STRING) {
-            long time = Long.parseLong(value.getString());
-            intervalBuilder.setStartTimeMillis(time);
-            intervalBuilder.setEndTimeMillis(time + TimeUnit.SECONDS.toMillis(periodSecs));
+            long startTime = Long.parseLong(value.getString());
+            long endTime = startTime + TimeUnit.SECONDS.toMillis(periodSecs);
+            intervalBuilder.setStartTimeMillis(startTime);
+            intervalBuilder.setEndTimeMillis(endTime);
 
             for (int i = idColumns.size() + 1;
                 i < chunk.getResultSetMetadata().getColumnMetadataCount();
                 i++) {
               ColumnMetadata metadata = chunk.getResultSetMetadata().getColumnMetadata(i);
+              // this where we have to add the values.
```

Comment:
What does this comment mean? Can we either remove or clarify?

Reply:
CC @kotharironak

Reply:
@aaron-steinfeld mostly this section of code is for adding the value to the response (`resultmap` in above code) which will be returned. Until here we were extracting some meta information about the data but at this point we find out the value from time aggregation (`row.getColumn(i)` in above code).

Reply:
@sarthak77 This comment is not required. I was just highlighting in draft PR where you have to make changes.